### PR TITLE
Cherry-pick #9563 to 6.5: Update golang 1.10.6

### PR DESCRIPTION
**Pull request by ph**
Cherry-pick of PR #9563 to 6.5 branch. Original message: 

Fix a few security issues:
```
cmd/go: remote command execution during "go get -u"
The issue is CVE-2018-16873 and Go issue golang.org/issue/29230. See the Go issue for details.
Thanks to Etienne Stalmans from the Heroku platform security team for discovering and reporting this issue.

cmd/go: directory traversal in "go get" via curly braces in import paths
The issue is CVE-2018-16874 and Go issue golang.org/issue/29231. See the Go issue for details.
Thanks to ztz of Tencent Security Platform for discovering and reporting this issue.

crypto/x509: CPU denial of service in chain validation
The issue is CVE-2018-16875 and Go issue golang.org/issue/29233. See the Go issue for details.
Thanks to Netflix for discovering and reporting this issue.
```

https://golang.org/doc/devel/release.html#go1.10.minor

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -33,6 +33,8 @@ https://github.com/elastic/beats/compare/v6.5.3...6.5[Check the HEAD diff]
 
 *Affecting all Beats*
 
+- Update Golang to 1.10.6. {pull}9563[9563]
```
Comment (graphaelli) [inline CHANGELOG.asciidoc]: do you normally change this to the backport pr number or leave it with the original?

Comment (ph) [reply CHANGELOG.asciidoc]: We leave to the original, its also referer in the backport commit message.

**Comment by ph** [comment]
jenkins test this

**Comment by tylerjl** [comment]
jenkins, test this please

**Pull request merged**